### PR TITLE
Added API endpoints for customer search and draft orders

### PR DESCRIPTION
**Pull request by akhtarja**
- `ShopifyAPI.REST.DraftOrder.*` - handling for draft orders in Shopify
- `ShopifyAPI.REST.Customer.search` - searches for customers that match a provided query string using Shopify's `customers/search.json` endpoint

### CODE REVIEW

File: lib/shopify_api/rest/customer.ex [hunk 1]
```diff
@@ -121,4 +121,15 @@ defmodule ShopifyAPI.REST.Customer do
...
+  @doc """
+  Search for customers that match a supplied query
+
+  ## Example
+
+      iex> ShopifyAPI.REST.Customer.search(auth, query_string)
+      {:ok, {"customers" => [] }}
+  """
+  def search(%AuthToken{} = auth, query_string),
+    do: Request.get(auth, "customers/search.json?query=#{query_string}")
```
Comment (hez) [inline lib/shopify_api/rest/customer.ex]: In the past we have done search filters like so https://github.com/pixelunion/elixir-shopifyapi/blob/master/lib/shopify_api/rest/asset.ex#L18

File: lib/shopify_api/rest/draft_order.ex [hunk 2]
```diff
@@ -0,0 +1,14 @@
+defmodule ShopifyAPI.REST.DraftOrder do
```
Comment (hez) [inline lib/shopify_api/rest/draft_order.ex]: 👍

File: lib/shopify_api/rest/draft_order.ex [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+  @moduledoc """
+  """
```
Comment (stewart) [inline lib/shopify_api/rest/draft_order.ex]: Could this have some descriptive content?

File: lib/shopify_api/rest/draft_order.ex [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+  alias ShopifyAPI.AuthToken
+  alias ShopifyAPI.REST.Request
+
+  @doc """
+    Create a new draft order
+
+  iex> ShopifyAPI.REST.DraftOrder.create(auth, %Order{})
```
Comment (stewart) [inline lib/shopify_api/rest/draft_order.ex]: There's a few styles of method documentation used in this library, I'd prefer if we started using something closer to other libraries, more in this style:

```elixir
@doc """
Creates a new DraftOrder with the provided attributes.

## Examples

    iex> ShopifyAPI.REST.DraftOrder.create(auth_token, %{draft_order: %{}})
    {:ok, %{"draft_order" => %{...}}}
"""

File: lib/shopify_api/rest/customer.ex [hunk 5]
```diff
@@ -121,4 +121,15 @@ defmodule ShopifyAPI.REST.Customer do
   """
   def get_orders(%AuthToken{} = auth, customer_id),
     do: Request.get(auth, "customers/#{customer_id}/orders.json")
+
+  @doc """
+  Search for customers that match a supplied query
+
+  ## Example
+
+      iex> ShopifyAPI.REST.Customer.search(auth, params)
```
Comment (stewart) [inline lib/shopify_api/rest/customer.ex]: Some possible search params may make this example more helpful.

File: lib/shopify_api/rest/customer.ex [hunk 6]
```diff
@@ -127,9 +127,9 @@ defmodule ShopifyAPI.REST.Customer do
...
   ## Example
 
-      iex> ShopifyAPI.REST.Customer.search(auth, query_string)
+      iex> ShopifyAPI.REST.Customer.search(auth, params)
       {:ok, {"customers" => [] }}
   """
-  def search(%AuthToken{} = auth, query_string),
-    do: Request.get(auth, "customers/search.json?query=#{query_string}")
+  def search(%AuthToken{} = auth, params),
+    do: Request.get(auth, "customers/search.json?" <> URI.encode_query(params))
```
Comment (hez) [inline lib/shopify_api/rest/customer.ex]: 👍

File: lib/shopify_api/rest/draft_order.ex [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+      iex> ShopifyAPI.REST.DraftOrder.complete(auth, integer, true)
+
+      {:ok, %{"draft_order", %{...}}}
+  """
+  def complete(%AuthToken{} = auth, draft_order_id),
+    do: Request.put(auth, "draft_orders/#{draft_order_id}/complete.json", %{})
+
+  def complete(%AuthToken{} = auth, draft_order_id, true),
+    do:
+      Request.put(auth, "draft_orders/#{draft_order_id}/complete.json?payment_pending=true", %{})
```
Comment (hez) [inline lib/shopify_api/rest/draft_order.ex]: This should probably just be merged with the one above and allow for filter params to be passed in.

File: lib/shopify_api/rest/draft_order.ex [hunk 8]
```diff
@@ -0,0 +1,141 @@
...
+      iex> ShopifyAPI.REST.DraftOrder.complete(auth, integer, true)
+
+      {:ok, %{"draft_order", %{...}}}
+  """
+  def complete(%AuthToken{} = auth, draft_order_id, payment_pending),
+    do:
+      Request.put(
+        auth,
+        "draft_orders/#{draft_order_id}/complete.json?" <>
+          URI.encode_query(%{"payment_pending" => payment_pending}),
```
Comment (hez) [inline lib/shopify_api/rest/draft_order.ex]: we should rename payment_pending  to params

Comment (akhtarja) [reply lib/shopify_api/rest/draft_order.ex]: I went back and forth on that, as this endpoint accepts only this one value as a parameter. I'll open a new PR for this, as this one is closed.

**Pull request merged**